torch/distributed/tensor/parallel/style.py

```diff
@@ -496,22 +457,22 @@ def __init__(
 
         if _prepare_input is not None:
             prepare_input_fn = _prepare_input
-        if input_layouts == Shard(-1):
-            prepare_input_fn = make_input_shard_1d_last_dim
         else:
-            prepare_input_fn = _get_prepare_input(
+            prepare_input_fn = functools.partial(
+                lambda input_layouts, output_layouts, inputs, device_mesh: (
```

Comment:
lambda might make things hard to debug, if the only reason of this lambda is to extract `inputs[0]`, can we do this inside functions like `_redistribute_per_both_layouts`?

Reply:
sure, added them as a small static method within the class.